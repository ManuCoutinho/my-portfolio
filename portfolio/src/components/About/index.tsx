import { Container, Title } from '../Foundation/base'
import { AboutSection } from './styles'
export const About: React.FC = () => {
  return (
    <Container>
      <AboutSection>
        <Title>Sobre</Title>
      </AboutSection>
    </Container>
  )
}
