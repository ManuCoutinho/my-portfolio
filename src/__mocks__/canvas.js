HTMLCanvasElement.prototype.getContext = () => {
  return {
    fillRect: jest.fn()
  }
}
