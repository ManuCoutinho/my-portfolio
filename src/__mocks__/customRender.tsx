import { ReactElement, FC, ReactNode } from 'react'
import { render } from '@testing-library/react'
import { RenderOptions } from '@testing-library/react/types'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
	return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
	ui: ReactElement,
	options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
