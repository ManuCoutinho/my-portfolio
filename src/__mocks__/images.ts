export const mockImg = [
	{
		imgPng: '/img1.png',
		imgWebp: '/img1.webp',
		alt: 'img1',
		id: 'img1'
	},
	{
		imgPng: '/img2.png',
		imgWebp: '/img2.webp',
		alt: 'img2',
		id: 'img2'
	}
]
