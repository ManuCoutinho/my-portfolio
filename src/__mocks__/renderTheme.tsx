import { ReactElement, ReactNode, FC } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
