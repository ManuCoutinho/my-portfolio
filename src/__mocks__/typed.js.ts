/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-empty-function */

export default class Typed {
  constructor(elementId: string, options: any) {}
  toggle(): void {}
  stop(): void {}
  start(): void {}
  destroy(): void {}
  reset(restart?: boolean): void {}
}
