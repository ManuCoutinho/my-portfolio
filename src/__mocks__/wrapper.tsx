/* eslint-disable react/display-name */
import { ThemeContextProvider } from 'contexts/ThemeContext'
import { DataProviderMock } from '__mocks__/providers/DataContext'

export const customWrapper = () => {
	return ({ children }: { children: React.ReactNode }) => (
		<ThemeContextProvider>
			<DataProviderMock>{children}</DataProviderMock>
		</ThemeContextProvider>
	)
}
