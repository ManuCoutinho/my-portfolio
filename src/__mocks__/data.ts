import { mockImg } from './images'
export default {
  description: 'a beutiful test',
  site: 'http://www.somesit.com',
  repo: 'http://www.somerepo.com',
  styles: 'styles',
  api: 'mock',
  framework: 'next',
  name: 'project',
  tools: ['tools'],
  id: 'id',
  doc: 'http://www.storybook.com',
  coverDefaultLight: 'img.png',
  coverDefaultDark: '',
  coverAltLight: '',
  coverAltDark: '',
  alt: 'mock img',
  img: mockImg,
  route: '/test'
}
