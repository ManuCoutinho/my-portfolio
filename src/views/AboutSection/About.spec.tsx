import { screen } from '@testing-library/react'
import { render } from '../../__mocks__/customRender'
import About from '.'

describe('<About />', () => {
	it('should render About section components correctly', () => {
		render(<About />)
		expect(screen.getByRole('heading', { name: /sobre/i })).toBeInTheDocument()
		expect(
			screen.getByRole('img', { name: /girl coding with cup of coffee/i })
		).toBeInTheDocument()
		expect(screen.getByRole('definition')).toBeInTheDocument()
		expect(screen.getByTestId('slider')).toBeInTheDocument()
	})
	it('should match to snapshot', () => {
		const { container } = render(<About />)
		expect(container.firstChild).toMatchSnapshot()
	})
})
