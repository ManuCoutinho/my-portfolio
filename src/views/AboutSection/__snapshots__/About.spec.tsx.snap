// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> should match to snapshot 1`] = `
<section
  class="sc-aadb450-0 dIZkPn"
  id="about"
>
  <h2
    class="sc-94e9b77b-0 dbunNU"
  >
    Sobre
  </h2>
  <div
    class="sc-54041553-0 kKTnzr"
  >
    <div
      class="sc-54041553-1 iuWuLo"
      data-state="inactive"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27400%27%20height=%27400%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="girl coding with cup in the left hand"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
    <div
      class="sc-ce9dac95-0 sc-9e6be24a-0 gbxaXB ihHBSK"
      data-state="inactive"
      data-testid="definition"
      direction="column"
    >
      <h3
        class="sc-94e9b77b-0 qYMbF"
      >
        <span>
          Olá, sou
        </span>
         Manu Coutinho!
      </h3>
      <p
        class="sc-f89260d2-0 eyKYlB"
      >
        Eu sou uma desenvolvedora front-end, movida por uma paixão em construir coisas capazes de integrar diversos universos por meio da tecnologia.
      </p>
      <p
        class="sc-f89260d2-0 eyKYlB"
      >
        Gosto de desenvolver interfaces dinâmicas, de alta fidelidade e com designs pensados para transferir a melhor experiência ao usuário final.
      </p>
      <p
        class="sc-f89260d2-0 eyKYlB"
      >
        Minha abordagem para a resolução de problemas é altamente inclusiva, colaborativa e centrada no ser humano.
      </p>
      <p
        class="sc-f89260d2-0 eyKYlB"
      >
        Aqui, compartillho alguns dos meus projetos!
      </p>
    </div>
  </div>
  <div
    class="sc-4f100a55-0 cSZsNv"
  >
    <h4
      class="sc-94e9b77b-0 qYMbF"
    >
      Minhas Skills:
    </h4>
    <div
      class="sc-4f100a55-1 jHNCwP keen-slider"
      data-testid="slider"
    >
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="akar-icons:html-fill"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="mdi:jquery"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="fa-brands:react"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="fa-brands:sass"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="ion:logo-javascript"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="mdi:nodejs"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="file-icons:storybook"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="file-icons:sentry"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="mdi:graphql"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="ph:figma-logo"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="simple-icons:testinglibrary"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="fa-brands:github"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="devicon-plain:docker-wordmark"
        />
      </div>
    </div>
    <div
      class="sc-4f100a55-1 jHNCwP keen-slider"
    >
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="file-icons:typescript-alt"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="file-icons:styledcomponents"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="file-icons:nextjs"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="file-icons:vite"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="simple-icons:jamstack"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="file-icons:tailwind"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="simple-icons:mui"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="simple-icons:chakraui"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="file-icons:jest"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="devicon-plain:githubactions"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="fa-brands:bootstrap"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="akar-icons:css-fill"
        />
      </div>
      <div
        class="sc-4f100a55-2 eIcsAz keen-slider__slide"
      >
        <iconify-icon
          class="sc-4f100a55-3 upgPG"
          classname="sc-4f100a55-3 upgPG"
          icon="fa-brands:git-alt"
        />
      </div>
    </div>
  </div>
</section>
`;
